### PR TITLE
Create view function accessors/example script

### PR DESCRIPTION
**Pull request by elliottdehn**
We need to create accessors for the view functions being added to Econia. I've implemented this such that the core function (`get_returns`) can be used to access any/all view functions that may be added now or in the future. It also allows the user to call a function themselves in the event of a bug occurring, which helps. However, using `get_returns` is somewhat of a challenge because the user has to convert their arguments to strings and parse the result. Therefore I've created helpers for all of the available view functions which do this for the user.

We _should_ also create examples for using both the entry functions and the view functions so that users can figure out how to get trading up-and-running (and more) quickly. I've created a script that starts from as much scratch as possible. It:
- Creates a market on an Econia given a deployed faucet (the one in our repo)--only if it doesn't exist already.
- Creates, initializes, registers, and deposits amounts in the faucet's two assets for two accounts.
- Performs market making activity in the form of limit orders on one account.
- Performs market taking activity in the form of market orders on the other acount.
- Offers some visibility into the change of state as a result via `get_price_levels` (a hotly-requested view function).

--

I've tested all of the view functions that return non-primitives (requiring parsing) by running them in the script by hand, given the state initialized by it. We have `get_returns` to stand in place of any other view functions which may have been screwed up. I can test each one by calling it if I have time but right now I think it's fine.

### CODE REVIEW

File: src/python/sdk/econia_sdk/view/market.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+          str(market_id),
+          str(market_order_id)
+        ]
+    )
+    return _convert_open_order_value(returns[0])
+
+def get_open_orders(
+    view: EconiaViewer,
+    market_id: int,
+    n_asks_max: int = 18446744073709551615,
```
Comment (alnoki) [inline src/python/sdk/econia_sdk/view/market.py:119]: Will you please change this from a magic number to a defined constant?

**Review by alnoki — CHANGES_REQUESTED** [review]
A few changes requested, also I advise rebasing onto `market-events` per #321

File: src/python/sdk/econia_sdk/view/market.py (hunk 1 above)
Comment (elliottdehn) [reply src/python/sdk/econia_sdk/view/market.py:119]: done

File: src/python/sdk/econia_sdk/view/market.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+def get_market_order_id_price(view: EconiaViewer, market_order_id: int) -> int:
+    returns = view.get_returns(
+        "market",
+        "get_market_order_id_price",
+        [],
+        [str(market_order_id)]
+    )
+    return int(returns[0])
+
+def get_market_order_id_side(view: EconiaViewer, market_order_id: int) -> bool:
```
[the remark this answers is not in the document]
Comment (elliottdehn) [reply src/python/sdk/econia_sdk/view/market.py:95]: done

File: src/python/sdk/econia_sdk/view/market.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        "custodian_id": int(value["custodian_id"]),
+        "market_id": int(value["market_id"]),
+        "market_order_id": int(value["market_order_id"]),
+        "price": int(value["price"]), # ticks per lot
+        "side": side,
+        "size": int(value["size"]), # lots of base
+        "user": AccountAddress.from_hex(value["user"])
+    }
+
+# get_open_orders_all skipped in lieu of above
```
[the remark this answers is not in the document]
Comment (elliottdehn) [reply src/python/sdk/econia_sdk/view/market.py:155]: done

File: src/python/sdk/econia_sdk/view/market.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+def get_price_levels(
+    view: EconiaViewer,
+    market_id: int,
+    n_ask_levels_max: int = 18446744073709551615,
```
[the remark this answers is not in the document]
Comment (elliottdehn) [reply src/python/sdk/econia_sdk/view/market.py:160]: done

File: src/python/sdk/econia_sdk/view/market.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+            "price": int(bid["price"]),
+            "size": int(bid["size"])
+        })
+    return {
+        "asks": asks,
+        "bids": bids,
+        "market_id": int(value["market_id"])
+    }
+
+# get_price_levels_all skipped in lieu of above
```
[the remark this answers is not in the document]
Comment (elliottdehn) [reply src/python/sdk/econia_sdk/view/market.py:192]: done

File: src/python/sdk/econia_sdk/view/user.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+        "get_market_account",
+        [],
+        [
+            serialize_address(user),
+            str(market_id),
+            str(custodian_id)
+        ]
+    )
+    return _convert_market_account_value(returns[0])
+
```
[the remark this answers is not in the document]
Comment (elliottdehn) [reply src/python/sdk/econia_sdk/view/user.py:83]: fixed. No not using an autolinter I don't think. Might figure out how to turn that on.

File: src/python/sdk/examples/trade.py [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+# mainnet: 0xc0deb00c405f84c85dc13442e305df75d1288100cdd82675695f6148c7ece51c
+# testnet: 0x40b119411c6a975fca28f1ba5800a8a418bba1e16a3f13b1de92f731e023d135
+# devnet: 0xc0de0000fe693e08f668613c502360dc48508197401d2ac1ae79571498cd8b74
+ECONIA_ADDR = AccountAddress.from_hex("0xc0de0000fe693e08f668613c502360dc48508197401d2ac1ae79571498cd8b74")
+FAUCET_ADDR = AccountAddress.from_hex("0x260d1327643a17897a8f28ac7279f93b49bef390fb70007b6a7d4456a6a32f29")
+COIN_TYPE_ETH = f"{FAUCET_ADDR}::test_eth::TestETH"
+COIN_TYPE_USDC = f"{FAUCET_ADDR}::test_usdc::TestUSDC"
+COIN_TYPE_APT = "0x1::aptos_coin::AptosCoin"
+NODE_URL = "https://fullnode.devnet.aptoslabs.com/v1"
+FAUCET_URL = "https://faucet.devnet.aptoslabs.com"
```
[the remark this answers is not in the document]
Comment (elliottdehn) [reply src/python/sdk/examples/trade.py:27]: turned this into a defaulted environment variable.

File: src/python/sdk/examples/trade.py [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+from econia_sdk.view.market import *
+from econia_sdk.view.registry import *
+from econia_sdk.view.resource_account import *
+from econia_sdk.view.user import *
+
+# mainnet: 0xc0deb00c405f84c85dc13442e305df75d1288100cdd82675695f6148c7ece51c
+# testnet: 0x40b119411c6a975fca28f1ba5800a8a418bba1e16a3f13b1de92f731e023d135
+# devnet: 0xc0de0000fe693e08f668613c502360dc48508197401d2ac1ae79571498cd8b74
+ECONIA_ADDR = AccountAddress.from_hex("0xc0de0000fe693e08f668613c502360dc48508197401d2ac1ae79571498cd8b74")
+FAUCET_ADDR = AccountAddress.from_hex("0x260d1327643a17897a8f28ac7279f93b49bef390fb70007b6a7d4456a6a32f29")
```
[the remark this answers is not in the document]
Comment (elliottdehn) [reply src/python/sdk/examples/trade.py:22]: turned this and econia address into environment variables.

File: src/python/sdk/examples/trade.py [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+from econia_sdk.types import *
+from econia_sdk.entry.market import *
+from econia_sdk.entry.user import *
+from econia_sdk.view.incentives import *
+from econia_sdk.view.market import *
+from econia_sdk.view.registry import *
+from econia_sdk.view.resource_account import *
+from econia_sdk.view.user import *
+
+# mainnet: 0xc0deb00c405f84c85dc13442e305df75d1288100cdd82675695f6148c7ece51c
```
[the remark this answers is not in the document]
Comment (elliottdehn) [reply src/python/sdk/examples/trade.py:18]: done

File: src/python/sdk/examples/trade.py [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+
+from econia_sdk.lib import EconiaViewer, EconiaClient;
+from econia_sdk.types import *
+from econia_sdk.entry.market import *
+from econia_sdk.entry.user import *
+from econia_sdk.view.incentives import *
+from econia_sdk.view.market import *
+from econia_sdk.view.registry import *
+from econia_sdk.view.resource_account import *
+from econia_sdk.view.user import *
```
[the remark this answers is not in the document]
Comment (elliottdehn) [reply src/python/sdk/examples/trade.py:16]: done

**Review by alnoki — CHANGES_REQUESTED** [review]
I'm not sure how this got rebased, but there are a few issues with file structure/git history

1. There shouldn't be any changes to `doc/doc-site/docs/move/changelog.md` or `src/move` (only changed files should be in `src/python/sdk`)
2. There should only be two commits here, reflecting the two diffs you made

**Comment by elliottdehn** [comment]
Closing for #364

**Pull request closed without merging**